### PR TITLE
CUDA: Remove some deprecated support, add CC 8.6 and 8.7

### PR DESCRIPTION
**Pull request by gmarkall**
This removes deprecated and various un-needed parts of the CUDA target, and adds support for Compute Capabilities 8.6 and 8.7 (dependent upon the toolkit version). The changes include:

- Remove support for CUDA toolkits 9.2 - 10.1.
- Test on gpuCI with 10.2, 11.1, 11.2, and 11.5:
   - 10.2: The oldest-supported, and only 10.x version supported
   - 11.1: The last version using NVVM 3.4
   - 11.2: The first version using NVVM 7.0
   - 11.5: The newest version supported on gpuCI (used for CUDA Python)
- Update the documentation to mention that CC 5.3 and CUDA Toolkit 10.2 are the minimum required.
   - Also update the supported device list to include newer devices like AGX Orin and H100.
   - Deprecation notice updated accordingly.
- Remove the 32-bit data layout, which has been unsupported for years.
- Remove vestigial parts of OSX support from the CUDA target (e.g. in `cuda_paths.py`).
- Update the libdevice search to not check for multiple virtual architectures (`compute_XX` etc) - these have not been used since CUDA 9.0.
- Remove asyncio support for Python < 3.7 (used in support for async operations on streams).
- Normalise the presentation of version numbers in `numba -s`:
   - Before: `(11, 6)` / `11050` for driver 11.6 with runtime 11.5
   - After: `11.6` / `11.5` for the same versions
- Always implement `math.remainder`, as it was only elided for Python < 3.7.
-  Bump the simulator reported CC to 5.3.
- Remove test skips for now-unsupported compute capabilities (<3.2, <5.0, and <5.3).

### CODE REVIEW

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
(I changed the suggested effort to Medium as this is mostly just a bunch of mechanical / straightforward changes and deletions)

**Review by stuartarchibald — DISMISSED** [review]
Thanks for the patch, changes look good.

**Comment by gmarkall** [comment]
Many thanks - is the buildfarm available for a run soon?

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
@stuartarchibald I've reinstated support for CC down to 3.5 in this PR, so it should still be fine with the buildfarm and your current hardware - the only new commit is https://github.com/numba/numba/pull/7878/commits/35b3cc81afe5b22c405bb397f3e4c46320b72ad6 - would you be able to give this a review and buildfarm run please?

File: docs/source/reference/deprecation.rst [hunk 1]
```diff
@@ -330,25 +330,24 @@ Schedule
...
 
-Deprecation of CUDA Toolkits < 10.2 and devices with CC < 5.3
-=============================================================
+Deprecation and removal of CUDA Toolkits < 10.2 and devices with CC < 5.3
+=========================================================================
 
-Support for:
+- Support for CUDA toolkits less than 10.2 was deprecated and removed.
+- Support for devices with Compute Capability < 5.3 is deprecated and will be
+  removed in future.
```
Comment (stuartarchibald) [inline docs/source/reference/deprecation.rst:338]:
```suggestion
  removed in the future.
```

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the update @gmarkall, there have been a few issues lately with users running on K40s so it seems a good idea to keep these as deprecated but functioning. There's a suggested minor wording fix else looks good.

**Comment by gmarkall** [comment]
gpuci run tests

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Comment by stuartarchibald** [comment]
Buildfarm ID: `numba_smoketest_cuda_yaml_124`.

**Comment by stuartarchibald** [comment]
> Buildfarm ID: `numba_smoketest_cuda_yaml_124`.

This passed on all build jobs that use the 11.2 CTK and failed on all that use 9.2 and 10.0. Given this is expected, will mark this as a pass, the buildfarm job config needs an update to test against a spread of newer CTKs.

**Pull request merged**